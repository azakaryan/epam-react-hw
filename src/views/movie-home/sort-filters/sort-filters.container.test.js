import React from 'react';
import { shallow } from 'enzyme';
import expectEnzymify from 'expect';
import enzymify from 'expect-enzyme';
import SortFilters, { mapDispatchToProps } from './sort-filters.containers';


expectEnzymify.extend(enzymify());
const onFiltersApply = () => {};

describe('Sort Filters Container Component', () => {
  it('should render correctly', () => {
    const component = shallow(<SortFilters onFiltersApply={onFiltersApply} />);
    expect(component).toMatchSnapshot();
  });

  it('should dispatch an action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).onFiltersApply();
    expect(dispatch.mock.calls[0][0]('', () => (new Map([['movies', []]])))).toEqual(undefined);
  });
});
