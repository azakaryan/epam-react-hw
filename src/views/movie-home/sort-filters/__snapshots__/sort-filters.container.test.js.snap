// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort Filters Container Component should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
